build.proj

```diff
@@ -10,6 +10,27 @@
 
   <Import Project="dir.traversal.targets" />
 
+  <PropertyGroup>
+    <TraversalBuildDependsOn>
+      BatchRestorePackages;
+      $(TraversalBuildDependsOn);
+    </TraversalBuildDependsOn>
+  </PropertyGroup>
+
+  <Target Name="BatchRestorePackages">
+    <Message Importance="High" Text="Restoring all packages..." />
+    <!-- restore all project.jsons in one pass for perf & to avoid concurrency problems with dnu -->
+    <Exec Command="$(DnuRestoreCommand) &quot;$(MSBuildProjectDirectory)\src&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
+
+    <ItemGroup>
+      <_allPackagesConfigs Include="$(MSBuildProjectDirectory)\src\**\packages.config"/>
+    </ItemGroup>
+    <Exec Condition="'@(_allPackagesConfigs)' != ''" Command="$(NugetRestoreCommand) &quot;%(_allPackagesConfigs.FullPath)&quot;" StandardOutputImportance="Low" />
+  </Target>
+
+  <!-- Override RestorePackages from dir.traversal.targets and do a batch restore -->
+  <Target Name="RestorePackages" DependsOnTargets="BatchRestorePackages" />
```

Comment:
I agree this should not happen in dir.traversal.targets. We do have to figure out it build.proj is the right place to do it as well however as we don't ever call build.proj from our internal builds. We could of course do it somewhere else for our internal builds as well but it would be nice to have it happen in a common place.